File: docs/snippets/all/archetypes/scalars_simple.rs
//! Log a scalar over time.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_scalar").spawn()?;

    // Log the data on a timeline called "step".
    for step in 0..64 {
        rec.set_time_sequence("step", step);
        rec.log("scalar", &rerun::Scalars::single((step as f64 / 10.0).sin()))?;
    }

    Ok(())
}

File: docs/snippets/all/archetypes/scalars_simple.cpp
// Log a scalar over time.

#include <rerun.hpp>

#include <cmath>

int main(int argc, char* argv[]) {
    const auto rec = rerun::RecordingStream("rerun_example_scalar");
    rec.spawn().exit_on_failure();

    // Log the data on a timeline called "step".
    for (int step = 0; step < 64; ++step) {
        rec.set_time_sequence("step", step);
        rec.log("scalar", rerun::Scalars(std::sin(static_cast<double>(step) / 10.0)));
    }
}
